src/Dapr.Workflow/WorkflowState.cs

```diff
@@ -0,0 +1,132 @@
+﻿// ------------------------------------------------------------------------
+// Copyright 2023 The Dapr Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//     http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ------------------------------------------------------------------------
+
+namespace Dapr.Workflow
+{
+    using System;
+    using Microsoft.DurableTask.Client;
+
+    /// <summary>
+    /// Represents a snapshot of a workflow instance's current state, including runtime status.
+    /// </summary>
+    public class WorkflowState
+    {
+        readonly OrchestrationMetadata? workflowState;
+
+        internal WorkflowState(OrchestrationMetadata? workflowState)
+        {
+            // This value will be null if the workflow doesn't exist.
+            this.workflowState = workflowState;
+        }
+
+        /// <summary>
+        /// Gets a value indicating whether the requested workflow instance exists.
+        /// </summary>
+        public bool Exists => this.workflowState != null;
+
+        /// <summary>
+        /// Gets a value indicating whether the requested workflow is in a running state.
+        /// </summary>
+        public bool IsWorkflowRunning => this.workflowState?.RuntimeStatus == OrchestrationRuntimeStatus.Running;
+
+        /// <summary>
+        /// Gets a value indicating whether the requested workflow is in a terminal state.
+        /// </summary>
+        public bool IsWorkflowCompleted => this.workflowState?.IsCompleted == true;
```

Comment:
Nit: Redundant `== true`?

Reply:
It's actually not redundant because of the `this.workflowState?.IsCompleted` resolves to `Nullable<bool>` and not `bool`.

Reply:
Sad, I thought .NET would figure that out for you.